client/theia-ecore/src/node/ecore-file-generation.ts

```diff
@@ -0,0 +1,108 @@
+/********************************************************************************
+ * Copyright (c) 2019 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import { ILogger } from "@theia/core";
+import { BackendApplicationContribution } from "@theia/core/lib/node/backend-application";
+import { RawProcess, RawProcessFactory } from "@theia/process/lib/node/raw-process";
+import { Application } from "express";
+import { inject, injectable } from "inversify";
+import * as path from "path";
+
+import { FileGenServer } from "../common/generate-protocol";
+
+
+@injectable()
+export class EcoreFileGenServer implements FileGenServer, BackendApplicationContribution {
+
+    constructor(
+        @inject(RawProcessFactory) protected readonly processFactory: RawProcessFactory,
+        @inject(ILogger) private readonly logger: ILogger) { }
+
+    generateEcore(name: string, prefix: string, uri: string, workspacePath: string): Promise<string> {
+        const jarPath = path.resolve(__dirname, '..', '..',
+            'src/server', 'ecore-backend-server-1.0-SNAPSHOT-jar-with-dependencies.jar');
+        if (jarPath.length === 0) {
+            throw new Error('The EcoreGeneration.jar is not found. ');
+        }
+
+        const command = 'java';
+        const args: string[] = [];
+
+        args.push(
+            '-jar', jarPath,
+            '-name', name,
+            '-prefix', prefix,
+            '-uri', uri,
+            '-workspacePath', workspacePath.substr(1)
```

Comment:
The `workspacePath.substr(1)` doesn't work in Linux, because you'd cut `/home/user/...` into `home/user/...`.
If I remove the `.substr(1)`, it works for me.

Reply:
On Windows when you remove the `.substr(1)`, the program throws the following error: `root ERROR Ecore Gen: java.nio.file.InvalidPathException: Illegal char <:> at index 2: /d:/Users/Uni/Dokumente/workspaces/EclipseSource/ecore-glsp/client/workspace\test.ecore
        at sun.nio.fs.WindowsPathParser.normalize(Unknown Source)
        at sun.nio.fs.WindowsPathParser.parse(Unknown Source)`

I wasn´t able to find another solution. Maybe you have an idea for another fix for this problem, that works on all platforms.

Reply:
Hm, not sure. If there is no nicer solution, the last resort could be in the Theia backend to just check for the OS with the [OS module](https://nodejs.org/api/os.html) -- `os.platform()`. For sure not nice, but would probably be the simplest fix.